rest/stats_context.go

```diff
@@ -235,7 +236,7 @@ func AddGoRuntimeStats() {
 }
 
 // Record Goroutines high watermark into expvars
-func recordGoroutineHighwaterMark(stats *expvar.Map, numGoroutines uint64) (maxGoroutinesSeen uint64) {
+func goroutineHighwaterMark(numGoroutines uint64) (maxGoroutinesSeen uint64) {
```

Comment:
I'm not completely convinced this fixes the problem, as it isn't clear what the original root cause was.

I'm wondering now if the actual issue is that base.MaxGoroutinesSeen isn't initialized to zero anywhere, and so LoadUint64 has the potential to return unpredictable results depending on the pointer address (which is why this might have been working for Sridevi originally?)

Reply:
So Ben and I had a look and believe we've found what the previous error was. We can see that inside of the `stats != nil` if statement `maxGoroutinesSeen` is set on the map. This would actually be the previous value of the max which would be 0 as this is what it is initialized to.

My fix worked because it was utilizing the returned value which was `numGoroutines`.

Reply:
If that were the case, wouldn't the value be updated eventually (and just lag by one logStats iteration), since it was doing atomic.StoreUint64(&base.MaxGoroutinesSeen, numGoroutines)?

Reply:
Discussed w/ Ben - he pointed out that it wouldn't be updated eventually if it only changes once, which could be the case for the QE test, given the logging frequency.